scripts/src/view/input/base.tsx

```diff
@@ -441,7 +448,9 @@ namespace fugazi.view.input {
 		public constructor(props: P, className: string, prompt?: string) {
 			super(props, "bgtext " + className, prompt);
 
-			this.state.backgroundText = props.backgroundText || "";
+			this.setState({
```

Comment:
you haven't moved it to `componentWillMount` but you changed it to use `setState`, why?